src/components/modals/Dialog.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useLayoutEffect } from 'react';
+import PropTypes from 'prop-types';
+import ReactDOM from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+import FocusTrap from 'focus-trap-react';
+import noScroll from 'no-scroll';
+import DenyIcon from '../../../lib/icons/react/DenyIcon';
+import Flex from '../layout/Flex';
+import FlexItem from '../layout/FlexItem';
+import Section from '../layout/Section';
+
+const Dialog = (props) => {
+  const handleClick = () => {
+    if (props.canDismiss) {
+      props.onDismiss();
+    }
+  };
+
+  const handleKeyDown = (e) => {
+    if (props.canDismiss && e.key === 'Escape') {
+      props.onDismiss();
+    }
+  };
+
+  useLayoutEffect(() => {
+    // This toggles scrolling on and off based on whether the modal
+    // is shown or not
+    if (props.isOpen) noScroll.on();
+    noScroll.off();
+  }, [props.isOpen]);
+
+  return ReactDOM.createPortal(
+    <CSSTransition timeout={200} in={props.isOpen} classNames="dialog" unmountOnExit>
+      <React.Fragment>
+        <FocusTrap>
+          <div>
```

Comment:
is this extra wrapping dev necessary?

Reply:
Yeah. I couldn't tell you why, but `<FocusTrap />` expects a single element, and it throws if you try to pass it a `<React.Fragment />`